crates/miden-objects/src/account/component/template/mod.rs

```diff
@@ -276,8 +276,17 @@ impl AccountComponentMetadata {
             }
         }
 
+        let mut seen_names = BTreeSet::new();
         for entry in self.storage_entries() {
             entry.validate()?;
+            if let Some(name) = entry.name() {
+                let name_existed = !seen_names.insert(name.as_str());
+                if name_existed {
+                    return Err(AccountComponentTemplateError::DuplicateEntryNames(
+                        name.as_str().into(),
+                    ));
+                }
+            }
```

Comment:
We should mention this error condition in the doc comments for this function.

Reply:
Updated validations and explanations